yetanotherco/aligned_layer
# feat(batcher): send response to clients once task has been confirmed in ethereum

~~**NOTE**: #337 should be merged first~~

Up to now, the batcher sent batch inclusion response to clients as soon as the batch verification task was created in Ethereum. Now, the responses are delayed until the `BatchVerified` event is emitted, meaning that the task was verified by operators and signatures validated in Ethereum. A timeout for the awaiting was also added so that if it is exceeded, the connection with the pending clients is closed gracefully.

To test it:
1. start anvil with block time, aggregator, operator and batcher
2. for simplicity, install the batcher client with `make install_batcher_client`
3. Without moving from the root of the repo, run
```
batcher-client --proving_system GnarkPlonkBls12_381 \
--proof batcher/client/test_files/plonk_bls12_381/plonk.proof \
--public_input batcher/client/test_files/plonk_bls12_381/plonk_pub_input.pub \
--vk batcher/client/test_files/plonk_bls12_381/plonk.vk \
--repetitions 2
```
You should receive a response once the batch signatures are verified in etherem.
4. To test the timeout, now stop the aggregator and operator
5. Send some more proofs with
```
batcher-client --proving_system GnarkPlonkBls12_381 \
--proof batcher/client/test_files/plonk_bls12_381/plonk.proof \
--public_input batcher/client/test_files/plonk_bls12_381/plonk_pub_input.pub \
--vk batcher/client/test_files/plonk_bls12_381/plonk.vk \
--repetitions 4
```
Note that we are sending 4 so that we don't get the error about batch already submitted
6. After a minute, an error will be promted in the client terminal with information about the timeout and the connection will be closed
